### PR TITLE
Eahw 2675/template repo

### PR DESCRIPTION
**Pull request by taomoh**

### CODE REVIEW

File: pipeline-scripts/check-build-promotion.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+LAST_SUCCESSFUL_BUILD=$(
```
Comment (taomoh) [inline pipeline-scripts/check-build-promotion.sh]: Move this script under bin/pipeline-scripts and update paths in .drone.yml accordingly

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		<version>3.0.4</version>
+		<relativePath/> <!-- lookup parent from repository -->
+	</parent>
+	<groupId>uk.gov.homeoffice.digital.sas</groupId>
+	<artifactId>REPLACE-SERVICE-NAME</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>REPLACE-SERVICE-NAME</name>
+	<description>Template for callisto services</description>
+	<properties>
+		<java.version>19</java.version>
```
Comment (taomoh) [inline pom.xml]: @stephenkaye-ho , do you foresee any issues with 19 not being an LTS version?

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+    file: Dockerfile
+
+steps:
+  - name: fetch
+    image: alpine/git
+    commands:
+      - "git fetch origin +refs/heads/main:"
+
+  - name: build
+    image: maven:3.8.3-openjdk-17
```
Comment (taomoh) [inline .drone.yml:106]: if we upgrade to jdk 19, change this accordingly

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+version: "3.3"
+
+services:
+  REPLACE-SERVICE-NAME:
+    image: maven:3.8.3-openjdk-17
```
Comment (taomoh) [inline docker-compose.yml:5]: if we upgrade to jdk 19, change this accordingly

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,6 @@
+FROM openjdk:19-alpine
```
Comment (taomoh) [inline Dockerfile]: ensure the same jdk version is used across the project

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+	<version>0.0.1-SNAPSHOT</version>
+	<name>REPLACE-SERVICE-NAME</name>
+	<description>Template for callisto services</description>
+	<properties>
+		<java.version>17</java.version>
+    <maven.compiler.source>${java.version}</maven.compiler.source>
+    <maven.compiler.target>${java.version}</maven.compiler.target>
+    <sonar.language>java</sonar.language>
+    <sonar.java.coveragePlugin>jacoco</sonar.java.coveragePlugin>
+    <checkstyle-maven-plugin.version>3.2.0</checkstyle-maven-plugin.version>
```
Comment (taomoh) [inline pom.xml:22]: Fix indentation

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,105 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (taomoh) [inline pom.xml:1]: Fix indentation in whole file

File: .drone.yml (hunk 3 above)
Comment (daleelR) [reply .drone.yml:106]: confirmed we're sticking to 17

File: docker-compose.yml (hunk 4 above)
Comment (daleelR) [reply docker-compose.yml:5]: confirmed we're sticking to 17

File: Dockerfile (hunk 5 above)
Comment (daleelR) [reply Dockerfile]: confirmed we're sticking to 17

File: .drone.yml [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+  event:
+    - pull_request
+
+steps:
+  - name: deploy_to_branch
+    image: pelotech/drone-helm3
+    settings:
+      namespace: callisto-dev
+      mode: upgrade
+      add_repos: test=https://ukhomeoffice.github.io/callisto-helm-charts/
```
Comment (taomoh) [inline .drone.yml]: Name repo something else, e.g helm-charts-repo

File: pipeline-scripts/check-build-promotion.sh (hunk 1 above)
Comment (daleelR) [reply pipeline-scripts/check-build-promotion.sh]: done

File: pom.xml (hunk 2 above)
Comment (daleelR) [reply pom.xml]: confirmed we're sticking to 17. has been updated

File: README.md [hunk 9]
```diff
@@ -1,2 +1,47 @@
...
+
+## Setup SonarCloud project
+
+Follow steps [here](https://collaboration.homeoffice.gov.uk/pages/viewpage.action?pageId=206901590) 
+
+## Drone secrets
+
+The following secrets will need adding to Drone. Details for finding secrets are in [Confluence](https://collaboration.homeoffice.gov.uk/display/EAHW/How+to).
+
+- `drone_token` - ??
```
Comment (joannafawl) [inline README.md]: I think we should remove the `??` from these unless we want to say if they need to be PR enabled or not

File: pom.xml (hunk 7 above)
Comment (daleelR) [reply pom.xml:1]: done

**Review by joannafawl — APPROVED** [review]

**Review by daleelR — APPROVED** [review]

**Pull request merged**